test/extensions/resource_monitors/injected_resource/config_test.cc

```diff
@@ -0,0 +1,38 @@
+#include "envoy/config/resource_monitor/injected_resource/v2alpha/injected_resource.pb.validate.h"
+#include "envoy/registry/registry.h"
+
+#include "common/event/dispatcher_impl.h"
+
+#include "server/resource_monitor_config_impl.h"
+
+#include "extensions/resource_monitors/injected_resource/config.h"
+
+#include "test/test_common/environment.h"
+#include "test/test_common/test_time.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ResourceMonitors {
+namespace InjectedResourceMonitor {
+
+TEST(InjectedResourceMonitorFactoryTest, CreateMonitor) {
+  auto factory =
+      Registry::FactoryRegistry<Server::Configuration::ResourceMonitorFactory>::getFactory(
+          "envoy.resource_monitors.injected_resource");
+  EXPECT_NE(factory, nullptr);
+
+  envoy::config::resource_monitor::injected_resource::v2alpha::InjectedResourceConfig config;
+  config.set_filename(TestEnvironment::temporaryPath("injected_resource"));
+  DangerousDeprecatedTestTime test_time;
+  Event::DispatcherImpl dispatcher(test_time.timeSource());
+  Server::Configuration::ResourceMonitorFactoryContextImpl context(dispatcher);
+  auto monitor = factory->createResourceMonitor(config, context);
+  EXPECT_NE(monitor, nullptr);
```

Comment:
the whole point of this test is to make sure that createResourceMonitor returns something non-null?

Also, how does this get deleted?  WDYT of using a concrete type rather than 'auto' for the type of 'monitor', because if that's a unique_ptr (or typedef to one) then I can see right away this doesn't leak.

Reply:
Yes, this test is just to exercise the very small amount of code in config.cc.

Changed monitor to a concrete type.